pkg/proxy/healthcheck/proxy_health.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package healthcheck
+
+import (
+	"fmt"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/events"
+	"k8s.io/klog/v2"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/utils/clock"
+	"net/http"
+	"time"
+)
+
+type ProxyHealthServer interface {
+	// Run starts the healthz HTTP server and blocks until it exits.
+	Run() error
+
+	// GetIPv4ProxierHealthUpdater returns IPv4 proxier health updater
+	GetIPv4ProxierHealthManager() ProxierHealthManager
+
+	// GetIPv6ProxierHealthUpdater returns IPv6 proxier health updater
+	GetIPv6ProxierHealthManager() ProxierHealthManager
+
+	proxierHealthChecker
+}
+
+var _ ProxyHealthServer = &proxyHealthServer{}
+
+// proxierHealthUpdater returns 200 "OK" by default. It verifies that the delay between
+// QueuedUpdate() calls and Updated() calls never exceeds healthTimeout.
+type proxyHealthServer struct {
+	listener    listener
+	httpFactory httpServerFactory
+	clock       clock.Clock
+
+	addr          string
+	healthTimeout time.Duration
+	recorder      events.EventRecorder
+	nodeRef       *v1.ObjectReference
+
+	ipv4ProxierHealthManager ProxierHealthManager
+	ipv6ProxierHealthManager ProxierHealthManager
```

Comment:
It might simplify things to have `ProxyHealthServer` just have `managers []ProxierHealthManager`, and the code that creates it would pass either one or two managers to `NewProxyHealthServer`?